Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandMoveLogic.cs

```diff
@@ -0,0 +1,112 @@
+            objDirectoin = CameraCache.Main.transform.InverseTransformDirection(objDirectoin);
+            handDirection = CameraCache.Main.transform.InverseTransformDirection(handDirection);
+
+            // Store the original rotation between the hand an object
+            m_gazeAngularOffset = Quaternion.FromToRotation(handDirection, objDirectoin);
+            m_draggingPosition = manipulationRoot.position;
+        }
+
+        /// <summary>
+        /// Updates gameobject with new position information of controller/hand
+        /// </summary>
+        /// <param name="centroid">center of translation to be used for Manipulation</param>
+        /// <param name="manipulationObjectPosition">position of gameobject to be manipulated</param>
+        /// <returns> a Vector3 describing the updated current Position of the gameObject being two-hand manipulated</returns>
+        public Vector3 Update(Vector3 centroid, Vector3 manipulationObjectPosition)
+        {
+            var newHandPosition = centroid;
+            var pivotPosition = GetHandPivotPosition();
+
+            // Compute the pivot -> hand vector in camera space
+            var newHandDirection = Vector3.Normalize(newHandPosition - pivotPosition);
+            newHandDirection = CameraCache.Main.transform.InverseTransformDirection(newHandDirection);
+
+            // The direction the object should face is the current hand direction rotated by the original hand -> object rotation.
+            var targetDirection = Vector3.Normalize(m_gazeAngularOffset * newHandDirection);
+            targetDirection = CameraCache.Main.transform.TransformDirection(targetDirection);
+
+            // Compute how far away the object should be based on the ratio of the current to original hand distance
+            var currentHandDistance = Vector3.Magnitude(newHandPosition - pivotPosition);
+            var distanceRatio = currentHandDistance / handRefDistance;
+            var distanceOffset = distanceRatio > 0 ? (distanceRatio - 1f) * DistanceScale : 0;
+            var targetDistance = objRefDistance + distanceOffset;
+
+            var newPosition = pivotPosition + (targetDirection * targetDistance);
+
+            var newDistance = Vector3.Distance(newPosition, pivotPosition);
+            if (newDistance > 4f)
+            {
+                newPosition = pivotPosition + Vector3.Normalize(newPosition - pivotPosition) * 4f;
+            }
+
+            m_draggingPosition = newPosition;
+
+            return m_draggingPosition;
+        }
+
+        ///<summary>
+        /// gets current controller/hand position
+        /// <returns>A point that is below and just in front of the head.</returns>
+        ///</summary>
+        public static Vector3 GetHandPivotPosition()
+        {
+            Vector3 pivot = CameraCache.Main.transform.position + new Vector3(0, -0.2f, 0) - CameraCache.Main.transform.forward * 0.2f; // a bit lower and behind
```

Comment:
let's define this `new Vector3(0, -0.2f, 0)` as a private readonly field so we don't have to create a new vector each time we get the hand pivot position.